python/dgl/frame.py

```diff
@@ -444,7 +444,7 @@ def pin_memory_(self):
         Does nothing if the storage is already pinned.
         """
         if not self.pinned_by_dgl and not F.is_pinned(self.data):
-            pin_memory_inplace(self.data)
+            self.data_nd = pin_memory_inplace(self.data)
```

Comment:
What if I do
```
g.ndata['x'] = torch.randn(...)
g.pin_memory_()
g.ndata['x'] = torch.randn(...)
```
`g.ndata['x']` will still be marked as `pinned_by_dgl = True`, but the actual content will not be pinned?

Reply:
This should be handled by the setter:
https://github.com/dmlc/dgl/blob/39721cedafa3caf69eb9e8b6bad9cc84a7dacf7f/python/dgl/frame.py#L241-L246
But you remind me I should delete `self.data_nd` in it.

Reply:
If I understand correctly, the snippets you gave will create a new Column to replace the old one instead of updating it.

Reply:
Added a unit test for it.